File: CS/WebSite/App_Code/GanttData.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.ComponentModel;

/// <summary>
/// Summary description for GanttData
/// </summary>
public class GanttData : DataSet
{
	public GanttData()
	{
        DataTable table = new DataTable("table");

        table.Columns.Add("GanntTask", typeof(string));
        table.Columns.Add("GanntStart", typeof(DateTime));
        table.Columns.Add("GanntEnd", typeof(DateTime));
        table.Columns.Add("GanttPercentageComplete", typeof(Int32));
        table.Columns.Add("GanttOwner", typeof(string));

        Tables.AddRange(new DataTable[] { table });
    }

    public static GanttData CreateData() {
        GanttData ds = new GanttData();
        DataTable table = ds.Tables["table"];

        table.Rows.Add(new object[] { "Analysing", new DateTime(2009,1,1), new DateTime(2009,1,5), 90,  "John"});
        table.Rows.Add(new object[] { "Planning", new DateTime(2009,1,5), new DateTime(2009,1,10), 80, "Williams" });
        table.Rows.Add(new object[] { "Implementation", new DateTime(2009,1,10), new DateTime(2009,1,25), 85, "Iverson" });
        table.Rows.Add(new object[] { "Maintaince", new DateTime(2009,1,25), new DateTime(2009,2,5), 90, "Young" });

        return ds;
    }

    #region Disable Serialization for Tables and Relations
    [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
    public new DataTableCollection Tables {
        get { return base.Tables; }
    }

    [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
    public new DataRelationCollection Relations {
        get { return base.Relations; }
    }
    #endregion Disable Serialization for Tables and Relations

}

File: VB/WebSite/App_Code/GanttData.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.ComponentModel

''' <summary>
''' Summary description for GanttData
''' </summary>
Public Class GanttData
    Inherits DataSet

    Public Sub New()
        Dim table As New DataTable("table")

        table.Columns.Add("GanntTask", GetType(String))
        table.Columns.Add("GanntStart", GetType(Date))
        table.Columns.Add("GanntEnd", GetType(Date))
        table.Columns.Add("GanttPercentageComplete", GetType(Int32))
        table.Columns.Add("GanttOwner", GetType(String))

        Tables.AddRange(New DataTable() { table })
    End Sub

    Public Shared Function CreateData() As GanttData
        Dim ds As New GanttData()
        Dim table As DataTable = ds.Tables("table")

        table.Rows.Add(New Object() { "Analysing", New Date(2009,1,1), New Date(2009,1,5), 90, "John"})
        table.Rows.Add(New Object() { "Planning", New Date(2009,1,5), New Date(2009,1,10), 80, "Williams" })
        table.Rows.Add(New Object() { "Implementation", New Date(2009,1,10), New Date(2009,1,25), 85, "Iverson" })
        table.Rows.Add(New Object() { "Maintaince", New Date(2009,1,25), New Date(2009,2,5), 90, "Young" })

        Return ds
    End Function

    #Region "Disable Serialization for Tables and Relations"
    <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
    Public Shadows ReadOnly Property Tables() As DataTableCollection
        Get
            Return MyBase.Tables
        End Get
    End Property

    <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)> _
    Public Shadows ReadOnly Property Relations() As DataRelationCollection
        Get
            Return MyBase.Relations
        End Get
    End Property
    #End Region ' Disable Serialization for Tables and Relations

End Class



